zcash_primitives/src/transaction/mod.rs

```diff
@@ -527,107 +566,49 @@ impl Transaction {
 
         Ok(())
     }
+
+    pub fn write_transparent<W: Write>(&self, mut writer: W) -> io::Result<()> {
+        if let Some(bundle) = &self.transparent_bundle {
+            Vector::write(&mut writer, &bundle.vin, |w, e| e.write(w))?;
+            Vector::write(&mut writer, &bundle.vout, |w, e| e.write(w))?;
+        } else {
+            CompactSize::write(&mut writer, 0)?;
+            CompactSize::write(&mut writer, 0)?;
+        }
+
+        Ok(())
+    }
+
+    #[cfg(feature = "zfuture")]
+    pub fn write_tze<W: Write>(&self, mut writer: W) -> io::Result<()> {
+        if let Some(bundle) = &self.tze_bundle {
+            Vector::write(&mut writer, &bundle.vin, |w, e| e.write(w))?;
+            Vector::write(&mut writer, &bundle.vout, |w, e| e.write(w))?;
+        } else {
+            CompactSize::write(&mut writer, 0)?;
+            CompactSize::write(&mut writer, 0)?;
+        }
+
+        Ok(())
+    }
```

Comment:
Slightly odd that most TZE transaction format parsing and serialization is removed, but this is left here unused.

Reply:
We no longer have a specification for where the TZE bits of the transaction should go! ZIP-222 needs to be updated, then this code will come back to life.